### ISSUE TITLE
Consider using semantic versioning for fluent projects

### ISSUE THREAD

**Issue by 6a68** [projectfluent/fluent#200]
Projects published via npm generally use [semantic versioning](https://docs.npmjs.com/getting-started/semantic-versioning) to encode meaning in the version number. Here's a short summary from semver.org:

> Given a version number MAJOR.MINOR.PATCH, increment the:
>
> MAJOR version when you make incompatible API changes,
> MINOR version when you add functionality in a backwards-compatible manner, and
> PATCH version when you make backwards-compatible bug fixes.

One of the benefits of semver for node projects is that consumers can specify loose version numbers, allowing for patch, minor, or major-level changes to be automatically installed when they are published.

I'm not sure what approach is used by fluent, but it seems that backward-incompatible changes are indicated by bumping the 'minor' version number, and both bug fixes and new, backward-compatible functionality fall under the 'patch' version number.

As a Mozilla-internal consumer of fluent libraries, I have learned to look a bit more closely at the changelog when fluent libraries are updated, as it doesn't conform to the semver standard. Folks who are external to Mozilla, or new to the fluent libraries, and who specify fuzzy versions in their package.json files, might be unpleasantly surprised to find that a 'minor' version update actually broke something.

Altogether, if fluent were to follow the semver standard, it would only help uptake and simplify maintenance tasks for consumers of the fluent libraries.

**Comment by 6a68** [comment]
Doh! Thought I had submitted this comment last week. Sorry for the delay

@stasm Hey, thanks for the detailed and thoughtful response. Although I do think the long-lived 0.x approach you describe is less common these days, it is still used by a few of the [most-depended npm packages](https://www.npmjs.com/browse/depended); and with fluent nearing version 1.0 (I wasn't aware of that, congrats :+1:), there's nothing to do here.

> Do you think this approach works well or has it been hard to follow Fluent's updates because of it? I'm always happy to help with making changes to the code when updates are non-trivial.

As a Mozilla project, we've definitely benefited from your help with major version updates. In fact, looking at [the history of fluent commits in Screenshots](https://github.com/mozilla-services/screenshots/search?q=fluent&type=Commits), your PRs have been the major fluent updates that we've landed. The current approach has worked fine for us, and I'll just plan to stick with that.

I'm going to go ahead and close this bug. Thanks again 👍
